### PR TITLE
Add depth to clone command

### PR DESCRIPTION
**Pull request by steelbrain**
Not everyone wants a full repo history just to compile.

[Review on Reviewable](https://reviewable.io/reviews/mongodb/libbson/137)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ sudo yum install mingw64-gcc automake autoconf libtool
 ### From Git
 
 ```sh
-git clone git://github.com/mongodb/libbson.git
+git clone git://github.com/mongodb/libbson.git -depth=1
```
Comment (bjori) [inline README.md]: Typo alert, that should be --depth=1

**Comment by bjori** [comment]
If you don't want the history then I don't recommend you do a git checkout, but rather fetch the actual releases, or the generated archives (e.g. https://github.com/mongodb/libbson/archive/master.zip)
I don't think recommending shallow clones is a good idea.

**Comment by steelbrain** [comment]
Well, I like git checkouts of repos instead of zip because if they update, I can just do git pull instead of downloading the whole zip of the same size. and I am sure I am not alone.

**Comment by bjori** [comment]
That is a good usecse for using --depth=1 indeed, but I'm remain skeptical of recommending it as the primary form of retrieving the sources in the main instructions which has a wide target audience.

People that are maybe more experienced, such as yourself, will always take the approach they are most comfortable with and fits their workflow and usecase. I'm sure there are plenty other neat tricks people could do to optimize the initial, or subsequent, downloads.

I think we need to stick to "general purpose" audience when it comes to these fundamental tasks as fetching the source code.

**Pull request closed without merging**